library/Graphite/Grapher.php

```diff
@@ -47,19 +53,65 @@ public function has(MonitoredObject $object)
         return true;
     }
 
+    public function parseGrapherConfig($graphite_vars = array())
+    {
+
+	if (!empty($graphite_vars)) {
+
+		if (!empty($graphite_vars->iframe_w)) {
+	    		$this->iframeWidth = $graphite_vars->iframe_w;
+		}
+		if (!empty($graphite_vars->iframe_h)) {
+	    		$this->iframeHeight = $graphite_vars->iframe_h;
+		}
+		if (!empty($graphite_vars->area_mode)) {
+	    		$this->areaMode = $graphite_vars->area_mode;
+		}
+
+	}
+
+    }
+
+    public function getKeysAndLabels($graphite_vars = array(), $vars) 
+    {
+
+        if (array_key_exists("graphite_keys", $vars)) {
+            $this->graphiteKeys = $vars["graphite_keys"];
+            $this->graphiteLabels = $vars["graphite_keys"];
+            if (array_key_exists("graphite_labels", $vars)) {
+		if (count($vars["graphite_keys"]) == count($vars["graphite_labels"])) {
+                    $this->graphiteLabels = $vars["graphite_labels"];
+                }
+            }
+        }
+
+    }
+
+    public function getPerfdataKeys($object) 
+    {
+    	foreach (PerfdataSet::fromString($object->perfdata)->asArray() as $pd) {
+            $this->graphiteKeys[] = $pd->getLabel();
+            $this->graphiteLabels[] = $pd->getLabel();
+        }
+    }
+
     public function getPreviewHtml(MonitoredObject $object)
     {
         $perfdata_property = $object->getType() . "_process_perfdata";
         if ( ! $object->$perfdata_property ) return '';
 
         $object->fetchCustomvars();
-        if (array_key_exists("graphite_keys", $object->customvars))
-            $graphiteKeys = $object->customvars["graphite_keys"];
-        else {
-            $graphiteKeys = array();
-            foreach (PerfdataSet::fromString($object->perfdata)->asArray() as $pd)
-                $graphiteKeys[] = $pd->getLabel();
-        }
+
+        if (array_key_exists("graphite", $object->customvars)) {
```

Comment:
The check is done in the new function.  This is 100% backward compatible in its current state.  I didn't change the keys and label at all.